src/Command/NotificatorProcessor.php

```diff
@@ -195,30 +204,25 @@ private function sendNotification(string $deviceToken, string $title, string $bo
     {
         $factory = $this->factory();
         $messaging = $factory->createMessaging();
-
-//        $notification = Notification::fromArray([
-//            'title' => $title,
-//            'body' => $body,
-//            'sound' => true,
-//        ]);
-
         $notification = Notification::create($title, $body);
-
         $message = CloudMessage::withTarget('token', $deviceToken)
             ->withNotification($notification);
 
         $result = $messaging->validateRegistrationTokens($deviceToken);
 
-        if (!empty($result['valid'])) {
-            $messaging->send($message);
+        try {
+            $messaging->send( $message );
+        } catch ( NotFound $e ) {
+            $message_error = $e->getMessage();
+            $message_mail = $message_error . 
+                            'Id устройства - ' . $device->getId();
+            mail( 'damodara16108@gmail.com', 'Error Notification', $message_mail );
+
+            // Если я правильно понимаю, то здесь нужно удалять девайс, т. к. ошибка говорит,
+            // что Firebase неизвестен этот токен
```

Comment:
use     
$this->database->remove($device);
and final commit to database with 
$this->database->flush();

Reply:
recomendation: 
in method initNotification(), create flag like `$flagDeviceWasDeleted = false;` before sendNotification() calls. 
Now, method sendNotification() have void type of return. Change it to bool and return true in case when you delete device.
and after all calls of sendNotofication() you can use 
```
        if ($flagDeviceWasDeleted) {
            $this->database->flush();
        }
```

Reply:
Сделано